docs/bigquery/snippets.py

```diff
@@ -871,67 +1078,114 @@ def test_copy_table_cmek(client, to_delete):
+    to_delete.append(bucket)
 
-    bucket_name = 'extract_person_ages_job_{}'.format(_millis())
-    # [START extract_table]
-    from google.cloud.storage import Client as StorageClient
+    # [START bigquery_extract_table]
+    # client = bigquery.Client()
+    # bucket_name = 'my-bucket'
+    destination_uri = 'gs://{}/{}'.format(bucket_name, 'shakespeare.csv')
+    dataset_ref = client.dataset('samples', project='bigquery-public-data')
+    table_ref = dataset_ref.table('shakespeare')
 
-    storage_client = StorageClient()
+    extract_job = client.extract_table(
+        table_ref, destination_uri)  # API request
+    extract_job.result()  # Waits for job to complete.
+    # [END bigquery_extract_table]
+
+    blob = bucket.get_blob('shakespeare.csv')
+    assert blob.exists
+    assert blob.size > 0
+    to_delete.insert(0, blob)
+
+
+def test_extract_table_json(client, to_delete):
+    from google.cloud import storage
+
+    bucket_name = 'extract_shakespeare_json_{}'.format(_millis())
+    storage_client = storage.Client()
     bucket = storage_client.create_bucket(bucket_name)  # API request
-    destination_blob_name = 'person_ages_out.csv'
-    destination = bucket.blob(destination_blob_name)
+    to_delete.append(bucket)
+
+    # [START bigquery_extract_table_json]
+    # client = bigquery.Client()
+    # bucket_name = 'my-bucket'
+    destination_uri = 'gs://{}/{}'.format(bucket_name, 'shakespeare.json')
+    dataset_ref = client.dataset('samples', project='bigquery-public-data')
+    table_ref = dataset_ref.table('shakespeare')
+    job_config = bigquery.job.ExtractJobConfig()
+    job_config.destination_format = (
+        bigquery.DestinationFormat.NEWLINE_DELIMITED_JSON)
 
-    destination_uri = 'gs://{}/{}'.format(bucket_name, destination_blob_name)
     extract_job = client.extract_table(
-        table_ref, destination_uri)  # API request
-    extract_job.result(timeout=100)  # Waits for job to complete.
+        table_ref, destination_uri, job_config=job_config)  # API request
+    extract_job.result()  # Waits for job to complete.
+    # [END bigquery_extract_table_json]
 
-    got = destination.download_as_string().decode('utf-8')  # API request
-    assert 'Bharney Rhubble' in got
-    # [END extract_table]
+    blob = bucket.get_blob('shakespeare.json')
+    assert blob.exists
+    assert blob.size > 0
+    to_delete.insert(0, blob)
+
+
+def test_extract_table_compressed(client, to_delete):
+    from google.cloud import storage
+
+    bucket_name = 'extract_shakespeare_compress_{}'.format(_millis())
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(bucket_name)  # API request
     to_delete.append(bucket)
-    to_delete.insert(0, destination)
+
+    # [START bigquery_extract_table_compressed]
+    # client = bigquery.Client()
+    # bucket_name = 'my-bucket'
+    destination_uri = 'gs://{}/{}'.format(bucket_name, 'shakespeare.zip')
```

Comment:
`.gz` is the common extension for gzipped files.

Reply:
I was following the command-line example [here](https://cloud.google.com/bigquery/docs/exporting-data#compression), which is where this snippet will go. Should I update the command-line example to use `.gz` as well when I make the docs update?

Reply:
Ah good catch. I've mailed CL 188763757 to update the docs.